avm/src/vm/opcode.rs

```diff
@@ -3205,9 +3253,13 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
       }
       hand_mem = cond_handler.clone().run(hand_mem).await;
       while current < limit && hand_mem.read_fixed(CLOSURE_ARG_MEM_START) > 0 {
-        hand_mem = body_handler.clone().run(hand_mem).await;
+        if args[2] != NOP_ID {
+          hand_mem = body_handler.clone().run(hand_mem).await;
+        }
         current = current + 1;
-        hand_mem = cond_handler.clone().run(hand_mem).await;
+        if args[2] != NOP_ID {
+          hand_mem = cond_handler.clone().run(hand_mem).await;
+        }
```

Comment:
Instead of this, maybe just detect that the body is a no-op at the top like the `cond_handler` and just short-circuit by consuming the `seq` object? (set the iterator to the same as its limit value)